src/python/pants/backend/javascript/import_parser/rules.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Iterable
+
+import pkg_resources
+
+from pants.backend.javascript.subsystems import nodejs
+from pants.backend.javascript.subsystems.nodejs import NodeJSToolProcess
+from pants.backend.javascript.target_types import JSSourceField
+from pants.core.util_rules import source_files
+from pants.core.util_rules.source_files import SourceFiles, SourceFilesRequest
+from pants.engine.fs import CreateDigest, FileContent
+from pants.engine.internals.native_engine import Digest, MergeDigests
+from pants.engine.internals.selectors import Get
+from pants.engine.process import ProcessResult
+from pants.engine.rules import Rule, collect_rules, rule
+from pants.engine.unions import UnionRule
+from pants.util.ordered_set import FrozenOrderedSet
+
+
+@dataclass(frozen=True)
+class InstalledJavascriptImportParser:
+    digest: Digest
+
+
+def _content_for_file(filename: str) -> FileContent:
+    return FileContent(path=filename, content=pkg_resources.resource_string(__name__, filename))
+
+
+@rule
+async def install_javascript_parser() -> InstalledJavascriptImportParser:
+    input_digest = await Get(
+        Digest,
+        CreateDigest(
+            [
+                _content_for_file("package.json"),
+                _content_for_file("package-lock.json"),
+                _content_for_file("script.cjs"),
+            ]
+        ),
+    )
+    result = await Get(
+        ProcessResult,
+        NodeJSToolProcess,
+        NodeJSToolProcess.npm(
+            args=("clean-install",),
+            description="install @pants/javascript-source-import-parser modules.",
+            input_digest=input_digest,
+            output_directories=("node_modules",),
+        ),
+    )
```

Comment:
I would be interested to learn if this works properly now, if you wouldn't mind trying ;)

```suggestion
    result = await Get(
        ProcessResult,
        NodeJSToolProcess.npm(
            args=("clean-install",),
            description="install @pants/javascript-source-import-parser modules.",
            input_digest=input_digest,
            output_directories=("node_modules",),
        ),
    )
```

since #17947 the type will be properly inferred from the constructor call.

Reply:
Seemingly not consistently some instances work some dont. I have yet to understand why, but this example gives me
```console
E   pants.base.exceptions.RuleTypeError: /home/tobiasn/code/pants/src/python/pants/backend/javascript/import_parser/rules.py:47: Could not resolve type for `NodeJSToolProcess.npm` in module pants.backend.javascript.import_parser.rules.
E   
E   Return type annotation required for `npm` in /home/tobiasn/code/pants/src/python/pants/backend/javascript/subsystems/nodejs.py:70
```
in tests.